crates/typst/src/engine.rs

```diff
@@ -202,6 +327,61 @@ impl Sink {
     }
 }
 
+/// Checks if a given warning is suppressed given one span it has a tracepoint
+/// in. If one of the ancestors of the node where the warning occurred has a
+/// warning suppression decorator sibling right before it suppressing this
+/// particular warning, the warning is considered suppressed.
+fn check_warning_suppressed(
+    span: Span,
+    world: &dyn World,
+    identifier: &diag::Identifier,
+) -> bool {
+    let Some(file) = span.id() else {
+        // Don't suppress detached warnings.
+        return false;
```

Comment:
For such simple let-else, I generally prefer the single-line style (which rustfmt only does if the semicolon is removed.)

Reply:
With your other suggestion, the line got long enough for `rust-analyzer` to insist on breaking it regardless.